File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/贪心/_02_中等/LeetCode_0122.java
package com.kurumi.codeThoughts.贪心._02_中等;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 122. 买卖股票的最佳时机 II
 */
public class LeetCode_0122 {
    static class Solution {
        public int maxProfit(int[] prices) {
            int res = 0;
            for (int i = 1; i < prices.length; i++) {
                int tmp = prices[i] - prices[i - 1];
                if (tmp > 0)
                    res += tmp;
            }
            return res;
        }
    }
}

File: 卡码/代码随想录/C++/动态规划/dp/04-股票问题/LeetCode_0122.cpp
//122. 买卖股票的最佳时机 II
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int maxProfit(vector<int>& prices) {
		int dp[2]={-prices[0],0};
		for(int i=1;i<prices.size();i++){
			dp[0]=max(dp[0],dp[1]-prices[i]);
			dp[1]=max(dp[1],dp[0]+prices[i]);
		}
		return dp[1];
	}
};

int main(){
  system("pause");
  return 0;
}
